src/libsync/owncloudpropagator.h

```diff
@@ -214,6 +214,16 @@ class PropagatorCompositeJob : public PropagatorJob
     {
         foreach (PropagatorJob *j, _runningJobs)
             j->abort();
+
+        // Give running jobs max 5 seconds to finish on their own, otherwise
+        // continue (that might result in conflicts owncloud/client/issues/5949)
+        int retriesNo = 5;
+        for(int i = 0; i < retriesNo ; i++) {
+            if (_runningJobs.size() == 0) {
+                break;
+            }
+            Utility::sleep(1);
```

Comment:
Ohh, true, just tested and it stops the main thread, yep will do that with signals